accounts-db/src/account_locks.rs

```diff
@@ -5,14 +5,14 @@ use {
     solana_sdk::{
         message::AccountKeys,
         pubkey::Pubkey,
-        transaction::{TransactionError, MAX_TX_ACCOUNT_LOCKS},
+        transaction::{Result, TransactionError, MAX_TX_ACCOUNT_LOCKS},
     },
     std::{cell::RefCell, collections::hash_map},
 };
 
 #[derive(Debug, Default)]
 pub struct AccountLocks {
-    write_locks: AHashSet<Pubkey>,
+    write_locks: AHashMap<Pubkey, u64>,
     readonly_locks: AHashMap<Pubkey, u64>,
 }
```

Comment:
because these have the same type, all the functions that change them are basically the same. we could use an enum or hashmap reference and delete half of them, but i left it like this for your review if you prefer the way it is or have no preference